massgov/mayflower
# DP-22292 Accessibility fixes for pages within binders

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description

**Table of contents**
- Set up to loop focus within the overlay after the last focusable item while the overlay is open.
- Set up ESC key to close the overlay when it's open and set focus on the overlay open button in the main page.

**Page flipper**
- Hide the duplicate links from keyboard and screen reader users.
- Hide unfocusable contextual info from screen reader users.
- Replace the component container `div` with `nav` for correct semantics.
- Set up unorder list for  the page flipper items to indicate their association to each other.
- Add `aria-label` to each link with contextual info and linked page title.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-22292)
- [Github issue]()

## Steps to Test


Use https://github.com/massgov/openmass/pull/831 with this PR to test in Drupal.

1. Go to http://localhost:3000/patterns/05-pages-binder-page-internal/05-pages-binder-page-internal.html
2. Tab to navigate the table of contents overlay open button.
3. Hit ENTER to open the overlay.
4. Tab through focusable items to the last one, then hit TAB again.  Find the close (x) button gets focus. Focus doesn't move onto the main page elements. As long as the overlay is open, focus loops in the overlay.
5. While the overlay is open, hit ESC key.  It doesn't matter where focus is in the overlay.  Find the ESC key triggers to close the overlay and focus is on the overlay open button.
6. Tab through the page flipper component. Find there are only 3 items, this is part of, previous and next page links get focus.  The arrow buttons, which are linked to the previous and next pages as duplicates, don't get focus. Mouse users can still click the button links to go to their linked pages.
7. With inspect, check each page link's `aria-label`.  Find value contains contextual info:
      - **This page is part of** PAGE TITLE.
      - **Previous page,** PAGE TITLE,
      - **Next page,** PAGE TITLE
8. In screen width 620px and smaller, check the markup of the arrow buttons in inspect.  It has `tabindex="0"` and no `aria-hidden="true"`. With keyboard and screen readers, the button gets focus. The text label beside the buttons shown in screen size 621px and greater are now hidden with css, `display: none;`.

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
